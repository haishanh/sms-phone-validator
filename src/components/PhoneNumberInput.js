import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import isPhoneNumberValid from '../helpers/validatePhoneNumber';

class PhoneNumberInput extends Component {
  state = {
    value: '',
    isInputValid: false
  };

  static propTypes = {
    placeholder: PropTypes.string.isRequired,
    onValidationChange: PropTypes.func
  };

  static defaultProps = {
    onValidationChange: () => {}
  };

  handleOnChange = ev => {
    ev.preventDefault();
    const { value } = ev.target;
    const { isInputValid } = this.state;
    const isInputValidNext = isPhoneNumberValid(value);
    this.setState({ value, isInputValid: isInputValidNext });
    if (isInputValid !== isInputValidNext) {
      this.props.onValidationChange(isInputValidNext);
    }
  };

  render() {
    const { placeholder } = this.props;
    const { value } = this.state;
    return (
      <Fragment>
        <input
          type="tel"
          maxLength="11"
          placeholder={placeholder}
          value={value}
          onChange={this.handleOnChange}
        />
      </Fragment>
    );
  }
}

export default PhoneNumberInput;
