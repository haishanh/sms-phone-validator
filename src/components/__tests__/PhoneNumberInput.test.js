import React from 'react';
import PhoneNumberInput from '../PhoneNumberInput';
import { shallow } from 'enzyme';

const noop = () => {};

describe('<PhoneNumberInput />', () => {
  it('should render', () => {
    const wrapper = shallow(<PhoneNumberInput placeholder="mobile" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke onValidationChange correctly', () => {
    const onValidationChange = jest.fn();
    const fakeClickEvent = {
      preventDefault: noop,
      target: {
        value: '188'
      }
    };
    const wrapper = shallow(
      <PhoneNumberInput
        placeholder="mobile"
        onValidationChange={onValidationChange}
      />
    );
    wrapper.find('input').simulate('change', fakeClickEvent);
    expect(onValidationChange.mock.calls.length).toBe(0);
    fakeClickEvent.target.value = '18817597436';
    wrapper.find('input').simulate('change', fakeClickEvent);
    expect(onValidationChange.mock.calls.length).toBe(1);
    expect(onValidationChange.mock.calls[0][0]).toBe(true);
  });
});
