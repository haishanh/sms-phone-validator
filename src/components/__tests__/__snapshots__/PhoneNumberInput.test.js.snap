// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneNumberInput /> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhoneNumberInput
    onValidationChange={[Function]}
    placeholder="mobile"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <input
        maxLength="11"
        onChange={[Function]}
        placeholder="mobile"
        type="tel"
        value=""
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "maxLength": "11",
        "onChange": [Function],
        "placeholder": "mobile",
        "type": "tel",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <input
          maxLength="11"
          onChange={[Function]}
          placeholder="mobile"
          type="tel"
          value=""
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "maxLength": "11",
          "onChange": [Function],
          "placeholder": "mobile",
          "type": "tel",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
